### PR TITLE
Fix 500 on logged-out DomainMetadata API request

### PR DESCRIPTION
**Pull request by orangejenny**
http://manage.dimagi.com/default.asp?189972

@calellowitz 
cc @esoergel

### CODE REVIEW

**Comment by calellowitz** [comment]
:+1:

File: corehq/apps/api/domain_metadata.py [hunk 1]
```diff
@@ -50,6 +51,10 @@ class DomainMetadataResource(HqBaseResource):
     calculated_properties = fields.DictField()
     domain_properties = fields.DictField()
 
+    # using the default resource dispatch function to bypass our authorization for internal use
+    def dispatch(self, request_type, request, **kwargs):
+        return Resource.dispatch(self, request_type, request, **kwargs)
```
Comment (esoergel) [inline corehq/apps/api/domain_metadata.py]: Surely there's some auth needed here, does tastypie handle it?

Comment (czue) [inline corehq/apps/api/domain_metadata.py]: yeah is this not a security hole or how does it work exactly? I would have thought it should `401` or something, but maybe that happens in the super call?

Comment (calellowitz) [inline corehq/apps/api/domain_metadata.py]: @orangejenny may have a better answer when she gets back but tastypie will still do user authentication. this only bypasses the domain level authorization I implemented when we restricted APIs to standard and up

Comment (czue) [inline corehq/apps/api/domain_metadata.py]: aha, that makes sense. thanks

**Pull request merged**